### DISCUSSION TITLE
Using shakapacker alongside jsbundling-rails

### DISCUSSION THREAD

**Discussion by chrismanderson** [shakacode/shakapacker#227] [Q&A] [answered]
I've been using webpacker combined with https://github.com/reactjs/react-rails for years now. While that setup does work, it's always been more brittle than I'd like.

I'm currently upgrading a suite of apps to Rails 7, so I'm revisiting the JS story as I'm sure many are. I really like the view-helper aspect of `react-rails` especially for my use cases which isn't a full blown React app nor simple enough to rely on standard JS.

It seems to be that `react-rails` relies on `shakapacker` and cannot be used with `jsbundling-rails` - a bit of a bummer, but makes perfect sense.

My question is - can I use both of these JS bundling tools together? Or is that a silly idea? I'd like to have `shakapacker` received for React usage only, and keep other, smaller JS (think moment, etc) that I use in my `erb` templates managed through the simpler `jsbundling-rails`. 

I'm just getting started understanding the Rails 7 JS ecosystem so it's entirely possible this is a dumb idea, can't/shouldn't be done, etc. But whatever the answer is, understanding why would be helpful!

**Comment by justin808** [comment, marked as answer]
Hi @chrismanderson, it probably can be done, but it would require some work.

**Reply by h0jeZvgoxFepBQ2C** [reply]
Could you maybe give us some hints how/what work would have to be done?

We are having a big monorepo for our clients (custom folder for views/models/controllers which are loaded vua autoload_paths via ENV variables during startup) and we dont want to upgrade all =require statements in our normal application.js files (like app/assets/javascripts/custom/customer1/application.js and app/assets/javascripts/custom/customer1/application2.js f.e).

We updated the jsx part to shakapacker, but would like to keep the sprockets way mostly unchanged (due to huge amount of work needed to migrate all customer js/css files). but sprockets gets more and more problems (due to ultraslow sass-rails -> sassc dependency https://github.com/sass/sassc-ruby/issues/189#issuecomment-1224140307 f.e. ).

So we think about keeping jsbundling-rails and cssbundling-rails and shakapacker at the same time?
